### PR TITLE
Add xyz.ljoonal.neospeeps

### PR DESCRIPTION
**Pull request by ljoonal**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.
  - App licensed under AGPL3+. Deps + resources under various other open licenses
- [x] I am an upstream contributor to the project.
  - The author of the submitted application
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains. 
  - Yes, own ljoonal.xyz, though have only set redirection at <https://neos.ljoonal.xyz/peeps> instead of the neospeep subdomain
- [x] Any additional patches or files have been submitted to the upstream projects concerned.
  - No additional patches

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
> Any additional patches or files have been submitted to the upstream projects concerned.
>    No additional patches

The the appdata file is upstream too?

**Comment by ljoonal** [comment]
> The the appdata file is upstream too?

Ah, it is? I mean, I can move it back to origin, just didn't feel like it made sense to have it separated from the manifest file :/ Guess I'll add it back and switcheroo the git tag to a new commit & update the manifest with it.

**Comment by hfiguiere** [comment]
bot, build xyz.ljoonal.neospeeps

**Comment by ljoonal** [comment]
Huh, I'm guessing based on the CI that the `release` field isn't optional as some of the flatpak docs seemed to suggest? And also seems need to add to the manifest that it's x86_64 only, as I don't really have a way to test aarch64 and the test build seems to indicate that it indeed would require some work for it to function. If there's anything else obvious that I've still missed feel free to tell me, as setting this up has been quite a confusing process ^^'

File: xyz.ljoonal.neospeeps.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  env: 
+    CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER: clang
+    CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS: -C link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold
+    CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: clang,
+    CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_RUSTFLAGS: -C link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold
+modules:
+  - name: neos_peeps
+    buildsystem: simple
+    only-arches:
+      - x86_64
```
Comment (hfiguiere) [inline xyz.ljoonal.neospeeps.yml:29]: this doesn't do what you think it does. 

Also why do you want to limit to x86_64. If you don't build aarch64 surely NO ONE will test it.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: xyz.ljoonal.neospeeps.yml (hunk 1 above)
Comment (ljoonal) [reply xyz.ljoonal.neospeeps.yml:29]: https://buildbot.flathub.org/#/builders/7/builds/8201 seemed to indicate that building on aarch64 failed, so I thought it'd make more sense to not keep trying to build while it's known to be broken? And if it doesn't do what I think it does, browsing trough the other PRs, should I instead add a `flathub.json` file which defines it then?

Comment (hfiguiere) [reply xyz.ljoonal.neospeeps.yml:29]: that's what the documentation says. But why do you need "clang" as linker?

Comment (hfiguiere) [reply xyz.ljoonal.neospeeps.yml:29]: but again it's disappointing to ignore aarch64

Comment (ljoonal) [reply xyz.ljoonal.neospeeps.yml:29]: > But why do you need "clang" as linker?

I basically just followed https://github.com/flathub/org.freedesktop.Sdk.Extension.rust-stable whilst trying to figure out how the whole flatpak building process worked, and was already familiar with mold so thought I'd just go ahead and keep using it everywhere.
Is there an issue with using it?

> but again it's disappointing to ignore aarch64

I mean, true I guess, but I'm already struggling to manage distributing this to x64 only, and I've yet to have a single person say that they'd use an aarch64 build.

Comment (hfiguiere) [reply xyz.ljoonal.neospeeps.yml]: so basically you don't know. I'd just remove clang altogether. It's not needed to build anything with Rust, that will remove any variable.

File: xyz.ljoonal.neospeeps.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    buildsystem: simple
+    build-options:
+      env:
+        CARGO_HOME: /run/build/neos_peeps/cargo
+    build-commands:
+      - cargo --offline fetch --manifest-path Cargo.toml --verbose
+      - cargo --offline build --release --verbose
+      - install -Dm755 target/release/neos_peeps /app/bin/neos_peeps
+      - cp *.md /app/
+      - install -Dm644 static/xyz.ljoonal.neospeeps.appdata.xml /app/share/metainfo/xyz.ljoonal.neospeeps.appdata.xml
```
Comment (JakobDev) [inline xyz.ljoonal.neospeeps.yml:30]:
```suggestion
      - install -Dm644 static/xyz.ljoonal.neospeeps.appdata.xml /app/share/metainfo/xyz.ljoonal.neospeeps.metainfo.xml
```

**Comment by hfiguiere** [comment]
bot, build xyz.ljoonal.neospeeps

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (hfiguiere) [inline flathub.json:1]: how about we drop this and try again

File: xyz.ljoonal.neospeeps.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: neos_peeps
+finish-args:
+  # Needs to talk to the network:
+  - --share=network
+  # Wayland access
+  - --socket=wayland
```
Comment (hfiguiere) [inline xyz.ljoonal.neospeeps.yml:12]: does it really support wayland?

Comment (ljoonal) [reply xyz.ljoonal.neospeeps.yml:12]: It should, internally just using [egui](https://github.com/emilk/egui) which afaik should support it. Additionally just tested that it at least worksOnMyMachineTM, so bug reports welcome if it doesn't work ^^'

**Comment by barthalion** [comment]
bot, build xyz.ljoonal.neospeeps

**Pull request closed without merging**